pkg/cmd/repo/autolink/list/http.go

```diff
@@ -0,0 +1,56 @@
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+type AutolinkGetter struct {
+	HttpClient *http.Client
+}
+
+func NewAutolinkGetter(httpClient *http.Client) *AutolinkGetter {
```

Comment:
**Suggestion**

Personally I'd be inclined to remove this since it doesn't really provide any value over the already exposed public `AutolinkGetter` type.